src/app/guards.ts

```diff
@@ -47,3 +47,7 @@ export const isHtmlElementWithHighlight = (thing: unknown): thing is dom.HTMLEle
 
 export const isKeyOf = <O extends {[key: string]: unknown}>(obj: O, thing: unknown): thing is keyof O =>
   typeof thing === 'string' && thing in obj;
+
+export const isPortaled = (thing: unknown): thing is {portalName: string} => (
+  typeof thing === 'object' && thing != null && 'portalName' in thing
```

Comment:
technically this'll still match `{portalName: undefined}` or `{portalName: ''}`, you might want to throw a `&& thing.portalName` on the end. but also wouldn't it be just as easy to check `params?.portalName` inline?